### PR TITLE
guest_os_booting: add new case for seabios loader

### PR DESCRIPTION
**Pull request by meinaLi**
This PR automate:
VIRT-297115 - Boot vm with seabios loader

### CODE REVIEW

**Comment by meinaLi** [comment]
```
# avocado run --vt-type libvirt --test-runner=runner --vt-machine-type q35 guest_os_booting.seabios_loader
JOB ID     : cc5e8f63f9c957b4abae807e75379d9720db8eb2
JOB LOG    : /var/lib/avocado/job-results/job-2023-06-26T23.00-cc5e8f6/job.log
 (1/3) type_specific.io-github-autotest-libvirt.guest_os_booting.seabios_loader.positive_test.default_loader: PASS (41.49 s)
 (2/3) type_specific.io-github-autotest-libvirt.guest_os_booting.seabios_loader.negative_test.pflash_type: PASS (20.54 s)
 (3/3) type_specific.io-github-autotest-libvirt.guest_os_booting.seabios_loader.negative_test.invalid_loader: PASS (18.48 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /var/lib/avocado/job-results/job-2023-06-26T23.00-cc5e8f6/results.html
JOB TIME   : 82.15 s
```

**Comment by meinaLi** [comment]
Label: modular_redesign

**Review by cliping — APPROVED** [review]
LGTM

File: libvirt/tests/src/guest_os_booting/seabios_firmware/seabios_loader.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    bkxml = vmxml.copy()
+
+    try:
+        guest_os.prepare_os_xml(vm_name, loader_dict)
+        if use_file:
+            fd = os.open(loader_path, os.O_CREAT)
+            os.close(fd)
+        guest_os.check_vm_startup(vm, vm_name, error_msg)
+    finally:
+        bkxml.sync()
```
Comment (dzhengfy) [inline libvirt/tests/src/guest_os_booting/seabios_firmware/seabios_loader.py:43]: As you do start vm in Line 41, do you need to destory vm before restoring vm xml?

Comment (meinaLi) [reply libvirt/tests/src/guest_os_booting/seabios_firmware/seabios_loader.py:43]: Actually after testing the case, the status of guest will return to the shut off state. So I don't think it's necessary to destroy the vm again.

Comment (dzhengfy) [reply libvirt/tests/src/guest_os_booting/seabios_firmware/seabios_loader.py:43]: That is because the vm is destoryed by avocado-vt post precessing

**Review by dzhengfy — APPROVED** [review]
LGTM

**Pull request merged**